[Lee Kang Ting Marcus] iP

creativequotient opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

Commit:
Add basic functionality

Duke class: add printWithBorders() method

This method takes a string as input and prints it within borders before
and after the string.

Duke class: add while-loop within main() to provide basic functionality

The while-loop receives inputs from the user and echos it back.
If the string "bye" is given, the program terminates with a goodbye message.

Commit:
modified Duke.java to provide functionality for Level-2

Implemented a String array (named `records`) to contain user entries.
Since it can be assumed that there will be at most 100 entries, a fixed size
array initialised to 100 was implemented.

Modified the while-loop in Duke.main() to add entries to `records`.

Modified the while-loop in Duke.main() to list out entries in `records`.

Commit:
add Task.java and TaskStatus.java

Encapsulation of a task as an object (containing its description and
status).

TaskStatus provides enumeration of completion status of tasks
(currently, only either pending or completed).

Commit:
modify Duke.java to keep track of task completion status

Rename `records` to `tasks` so as to make naming more intuitive.

Change tasks from String[] to Task[] as tasks are now stored as Task
objects instead of String.

Modify program loop to enable listing of tasks with completion status
and also ability to mark tasks as complete.

Commit:
add Deadline.java, Todo.java, Event.java and TaskType.java

Add TaskType.java to provide enumerations for task types.

Add Deadline, Todo and Even classes which are sub-classes of Task to support different for the three types of entries.

Commit:
modify Duke.java and Task.java

Modify Duke.java to reduce repeated print statements by streamlining
the message generation step to create a new StringBuilder object before
the program-loop and then a single call to print out the message at the
end.

Modify Task.java to add method to parse user input in order to
determine the type of Task the user has entered and process them in their own
task-specific manner.

Commit:
modify Duke.java

remove redundant whitespace at EOL

Commit:
modify Duke.java

removed trailing whitespace at EOL

Commit:
modify text-ui-test files

Modify input.txt and EXPECTED.TXT to have updated test
cases in line with recent changes to the programme.

Commit:
modified Duke.java and EXPECTED.TXT

modified output of "list" command to also include the proper headers

Commit:
add DukeException.java

Commit:
modified Task and sub-classes to throw Exceptions

Modified Task.java, Deadline.java, Task.java and Todo.java to throw
DukeExceptions when parsing errors are encountered

Commit:
modify Duke.java to handle exceptions

Modify Duke.java to catch DukeExceptions thrown by methods in the
program-loop when errors/incorrect inputs are encountered. Also modify program loop
to throw DukeExceptions on unknown commands.

Commit:
modify Duke.java

Modify Duke.java to use an ArrayList to store tasks instead of Task[] array.

Commit:
modify Duke.java to support deletion of tasks

Modify Duke.java program-loop to support "delete" command

Commit:
modify Duke.java to display correct exception message

Commit:
modify input.txt and EXPECTED.txt

Modify test cases to support newly added functionality
such as exceptions and deletions.

Commit:
modify Task.java to use switch instead of if-else block

Commit:
modify Duke.java to throw exceptions on invalid options

Commit:
modify test cases to test new additions

modify test cases to test for proper behaviour when a "done" or
"delete" command is provided with a number that is invalid.

Commit:
Remove completed TODOs

Commit:
Add ability to save and reload tasks to /from file

Modify Duke, Task, Todo, Event and Deadline to save tasks to file and reload on program startup.

Previously, tasks were lost at the end of the program.

Tasks are now saved at the end of the program ("Bye" command) and
reloaded at the start of the program. To facilitate this, saveTasks and
loadTasks method were implemented in Duke class to save and load tasks. Task
(and sub-classes) also modified to now provide a storageEntry method to
output the string formatted entry to be written to file.

Commit:
Update runtest.sh

Modify runtest.sh to delete data/tasks.txt so as to prevent reloading
of tasks from previous tasks from interfering with the next tests.

Commit:
Modify Duke.java to update file on every command

Commit:
Modify Deadline.java to treat parse datetime objects

Deadline treated date or date and time as strings.

Modified Deadline class to store as LocalDate and LocalTime objects instead.

Commit:
Modify test cases

Modify test cases to match the new Deadline handling of datetime.

Commit:
Modify test cases

Update test cases to test newly added functionality with regards to
Deadline.java behavior when dealing with datetime objects.

Commit:
Add Parser, Storage, TaskList and Ui classes

Previously, the I/O, user-facing UI code, task list, etc was handled inside the Duke class.

This is a violation of OOP and these different functionalities can be
extracted to respective classes for easier maintenance and clarity.

All methods to do with user commands have now been extracted to UI. A new
TaskList class has been added to support methods such as get, size, remove, etc.
Storage class handles the IO read/write of tasks to and from disk. Finally, Parser
processes the user commands and dispatches the appropriate commands.

Commit:
Modify test cases

Update test cases to test newly added functionality added
with the previous commit.

Commit:
Reorganize code into packages

Commit:
add Unit tests and minor variable name changes

add Unit tests for Todo and Deadline
modify Event.java to amend "timing" to "location"

Commit:
add Javadoc for public classes and methods

Commit:
modify EXPECTED.TXT

Commit:
modify .java files in src/ to meet coding standards

Commit:
add "find" command support

There was previously no way for users to query the task list which is
not efficient if the user has a lot of tasks.

Support for a new command, "find", has now been added which displays
only tasks that meet the criteria.

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/tasks/Task.java

Commit:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/tasks/Deadline.java
#	src/main/java/duke/tasks/Event.java
#	src/main/java/duke/tasks/Task.java
#	src/main/java/duke/tasks/Todo.java

Commit:
refactor code to maintain code standards after merging

Commit:
add Javadoc to enum classes

Commit:
modify Parser to remove need for Storage obj

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

Commit:
modify Duke.java to remove duplicate method

Commit:
Merge branch 'add-gradle-support' into A-Gradle

Commit:
modify grade-wrapper.properties file

Commit:
add Checkstyle integration

Commit:
Merge branch 'branch-A-CheckStyle'

clarlzx commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,171 @@
+     * @return Corresponding Task object
+     */
+    public static Task parseRecord(String fullRecord) {
+        String[] recordSplit = fullRecord.split("\\|");
```
Perhaps a plural could be used for the variable name of this string array?

creativequotient replied:
Thank you for your comment, good suggestion!

clarlzx commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,107 @@
+import duke.ui.Ui;
+
+/**
+ * Parser is the class that parses raw user commands and executes the intended effect
```
Maybe end with a full stop for consistency for the description of classes and methods?

clarlzx commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,76 @@
+     * @param taskList List\<Task\> of tasks to be written to disk
+     * @throws DukeException If IOException encountered
+     */
+    public void save(List<Task> taskList) throws DukeException {
```
Maybe tasks will be a better name than taskList?

clarlzx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+
+        while (true) {
+            String userCommand = this.ui.getUserCommand();
+            boolean exit = this.parser.parse(userCommand);
```
Perhaps isExit would be a better name?

clarlzx commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,65 @@
+ * Ui class which handles interactions with the user
+ */
+public class Ui {
+    private static final String BORDER = "-------------------------------------";
```
I like that the name was capitalized to follow coding standards for constants. 😄

car155 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,129 @@
+     * @param time        that the task is due
+     * @param completed   <code>true</code> if the event is completed, <code>false</code> otherwise
+     */
+    public Deadline(String description, LocalDate date, LocalTime time, Boolean completed) {
+        super(description, TaskType.DEADLINE, completed);
+        this.date = date;
+        this.time = time;
+    }
```
Maybe use isCompleted instead of completed?

car155 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder output = new StringBuilder();
+        switch (this.type) {
+        case TODO:
+            output.append("[T]");
+            break;
+        case EVENT:
+            output.append("[E]");
+            break;
+        case DEADLINE:
+            output.append("[D]");
+            break;
+        }
+        output.append(this.isComplete() ? "[X] " : "[ ] ");
+        output.append(this.description);
+        return output.toString();
+    }
```
Good formatting for switch cases!

Commit:
add JavaFX gui functionality

Commit:
modify Main to allow named params

Specify the path to the file where tasks are stored with --data==file

Commit:
Merge branch 'branch-A-Varargs'

Commit:
add assert statements

Commit:
refactor code base for readability

Under the CLI model, Ui class contained a scanner which was needed to read in
user input.

In current GUI implementation, Ui class methods can be static because they are
functionally wrapper functions to decorate/format the strings accordingly.

Commit:
Merge pull request #1 from kangtinglee/branch-A-Assertions

add assert statements

Commit:
Merge pull request #2 from kangtinglee/branch-A-CodeQuality

refactor code base for readability

Commit:
add gradle CI

Commit:
Merge pull request #3 from kangtinglee/branch-A-CI

add gradle CI

Commit:
move gradle.yml CI file

Commit:
Merge pull request #4 from kangtinglee/branch-A-CI

move gradle.yml CI file

Commit:
rename github actions workflows dir

Commit:
Merge pull request #5 from kangtinglee/branch-A-CI

rename github actions workflows dir

Commit:
modify Parser for friendlier syntax

Modify Parser to utilize Regex to determine the kind of command given

Commit:
Merge pull request #6 from kangtinglee/branch-C-FriendlierSyntax

modify Parser for friendlier syntax

Commit:
modify GUI for for more aesthetic value

modify fxml files to make stylistic changes to GUi such as rounded borders,
gradients, resizing, etc.

add Response.java to encapsulate a response from Duke to be displayed by the
GUI which contains information about whether or not the previous command triggered
an error.

Commit:
Merge branch 'branch-A-BetterGUI'

Commit:
add new user and duke icons

Commit:
Merge branch 'branch-A-BetterGUI'

Commit:
add clear command

Commit:
add Command.java and add aesthetic changes

Previously, the execution of a given user command was handled by the
Parser class.

However, this violates SLAP and also results in `Parser#parseCommand`
being too deeply nested and untidy.

Add Command (and subclasses) class to handle the execution so that
`Parser#parseCommand` will only have 1 level of abstraction and move the execution of
commands into Command class (and subclasses).

Modify DialogBox.java and corresponding .fxml files to create two
separate .fxml files for DukeDialog and UserDialog for more control over
styling.

Commit:
add credit for code reuse

Commit:
add Ui.png

Commit:
update docs/README.md user guide

Commit:
Merge branch 'branch-A-UserGuide'

Commit:
update user guide with delete command

Commit:
update tasks.txt to match items in user guide
